src/k-provider/ovp/regex-action-handler.js

```diff
@@ -0,0 +1,147 @@
+    const regexAction = RegexActionHandler._extractRegexActionFromData(data);
+    if (regexAction && regexAction.pattern && regexAction.replacement) {
+      const regExp = new RegExp(regexAction.pattern, 'i');
+      if (cdnUrl.match(regExp)) {
+        cdnUrl = cdnUrl.replace(regExp, regexAction.replacement);
+        if (regexAction.checkAliveTimeoutMs && regexAction.checkAliveTimeoutMs > 0) {
+          let urlToPing = cdnUrl + '/api_v3/service/system/action/ping/format/1';
+          const finalMediaConfig = await RegexActionHandler._pingECDNAndReplaceUrls(mediaConfig, regexAction, urlToPing);
+          return finalMediaConfig;
+        } else {
+          RegexActionHandler._modifyUrls(mediaConfig, regexAction);
+          return mediaConfig;
+        }
+      }
+    } else {
+      return mediaConfig;
+    }
+  }
+
+  /**
+   * Modify the urls
+   * @function _modifyUrls
+   * @param {ProviderMediaConfigObject} mediaConfig - The media config
+   * @param {KalturaAccessControlModifyRequestHostRegexAction} regexAction - The regex action
+   * @returns {void}
+   * @static
+   * @private
+   */
+  static _modifyUrls(mediaConfig: ProviderMediaConfigObject, regexAction: KalturaAccessControlModifyRequestHostRegexAction) {
+    RegexActionHandler._logger.debug(`Starting to modify urls...`);
+    const sources = mediaConfig.sources;
+    if (sources.hls.length > 0) {
+      sources.hls.forEach(src => (src.url = RegexActionHandler._applyRegexAction(regexAction, src.url)));
+    }
+    if (sources.dash.length > 0) {
+      sources.dash.forEach(src => (src.url = RegexActionHandler._applyRegexAction(regexAction, src.url)));
+    }
+    if (sources.progressive.length > 0) {
+      sources.progressive.forEach(src => (src.url = RegexActionHandler._applyRegexAction(regexAction, src.url)));
+    }
+    if (sources.image.length > 0) {
+      sources.image.forEach(src => (src.url = RegexActionHandler._applyRegexAction(regexAction, src.url)));
+    }
+    if (OVPConfiguration.get().replaceECDNAllUrls) {
+      if (sources.captions.length > 0) {
+        sources.captions.forEach(src => (src.url = RegexActionHandler._applyRegexAction(regexAction, src.url)));
+      }
+      if (sources.poster) {
+        sources.poster = RegexActionHandler._applyRegexAction(regexAction, sources.poster);
+      }
+    }
+    RegexActionHandler._logger.debug(`Finished modifying urls`);
+  }
+
+  /**
+   * Extracts the regex action from the data response
+   * @function _extractRegexActionFromData
+   * @param {Map<string, Function>} data - The response data
+   * @returns {KalturaAccessControlModifyRequestHostRegexAction} regexAction - The regex action
+   * @static
+   * @private
+   */
+  static _extractRegexActionFromData(data: any): ?KalturaAccessControlModifyRequestHostRegexAction {
+    if (data.has(OVPMediaEntryLoader.id)) {
+      const mediaLoader = data.get(OVPMediaEntryLoader.id);
+      if (mediaLoader && mediaLoader.response) {
```

Comment:
i think that this can be merged into 
if (data.get(OVPMediaEntryLoader.id)?.response)

Reply:
not exactly because we use the mediaLoader in line 140. however, I can get everything shorter by doing something like:
```
if (data.has(OVPMediaEntryLoader.id)) {
      return data.get(OVPMediaEntryLoader.id)?.response?.playBackContextResult.getRequestHostRegexAction();
    }
```